openerp/addons/base/ir/ir_attachment.py

```diff
@@ -141,6 +141,18 @@ def _get_path(self, cr, uid, bin_data, sha):
             os.makedirs(dirname)
         return fname, full_path
 
+    def _mark_for_gc(self, cr, uid, fname):
+        """ Add ``fname`` in a checklist for the filestore garbage collection. """
+        # the checklist uses a spooldir pattern: add an empty file in the
+        # directory "checklist" of the filestore
+        full_path = os.path.join(self._full_path(cr, uid, "checklist"), fname)
+        if not os.path.exists(full_path):
+            dirname = os.path.dirname(full_path)
+            if not os.path.isdir(dirname):
+                with tools.ignore(OSError):
```

Comment:
Here I think only `EEXIST` should be ignored, not all OSError (e.g. if the access rights are incorrect or there's no more inode or whatever it probably shouldn't be ignored?)